src/js/view/controls/components/menu/menu.js

```diff
@@ -470,11 +476,17 @@ export default class Menu extends Events {
                 this.categoryButton.ui.destroy();
             }
             if (this.itemsContainer) {
+                emptyElement(this.itemsContainer.el);
                 this.itemsContainer.destroy();
             }
             if (this.topbarUI) {
                 this.topbarUI.destroy();
             } 
+            if (this.items) {
```

Comment:
Destroy just removes event listeners/references on an instance of ui.js. It doesn't remove the listeners on the containing element. Order doesn't really matter here, we just have to do both.

Reply:
Not sure you need the `if` here unless you set `this.items` to `null` in this block.

It looks like `this.items` is always an array, so if you want to be safe empty (`items.length = 0`) or reset (`items = []`) here. 

Might want to sub-function `this.items.forEach(item => item.destroy())` too since you do it in two places.

Reply:
Good call, I'll remove this/make it more DRY